src/tasks/solvers.ts

```diff
@@ -0,0 +1,138 @@
+import "hardhat-deploy";
+import "@nomiclabs/hardhat-ethers";
+
+import { Contract, Signer } from "ethers";
+import { subtask, task } from "hardhat/config";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+
+const solversTaskList = ["add", "check", "remove"] as const;
+type SolversTasks = typeof solversTaskList[number];
+
+async function getOwnerSigner({
+  ethers,
+  getNamedAccounts,
+}: HardhatRuntimeEnvironment): Promise<Signer> {
+  const { owner } = await getNamedAccounts();
+  const signer = (await ethers.getSigners()).find(
+    (signer) => signer.address == owner,
+  );
+  if (signer == undefined) {
+    throw new Error(
+      "No owner found among the signers. Did you export the owner's private key?",
+    );
+  }
+  return signer;
+}
+
+async function getAuthenticator({
+  ethers,
+  deployments,
+}: HardhatRuntimeEnvironment): Promise<Contract> {
+  const authenticatorDeployment = await deployments.get(
+    "GPv2AllowListAuthentication",
+  );
+
+  const authenticator = new Contract(
+    authenticatorDeployment.address,
+    authenticatorDeployment.abi,
+  ).connect(ethers.provider);
+
+  return authenticator;
+}
+
+async function addSolver(
+  solver: string,
+  hardhatRuntime: HardhatRuntimeEnvironment,
+) {
+  const owner = await getOwnerSigner(hardhatRuntime);
+  const authenticator = await getAuthenticator(hardhatRuntime);
+
+  const tx = await authenticator.connect(owner).addSolver(solver);
```

Comment:
ubernit: these three lines are the same in all commands, maybe they can be put into `getConnectedAuthenticator`?

Reply:
I like this suggestion very much!

Reply:
Two commands look like this (add and remove), one (check) does not need the owner. Since there is this difference and the distinction points out when the owner is needed, I think it's ok to have these lines repeat twice.